config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java

```diff
@@ -107,13 +107,18 @@ class MpConfigImpl implements Config {
 
     @Override
     public ConfigValue getConfigValue(String key) {
+
+        ConfigValue value = findConfigValue(key)
+                .orElseGet(() -> new ConfigValueImpl(key, null, null, null, 0));
+
         if (configProfile == null) {
-            return findConfigValue(key)
-                    .orElseGet(() -> new ConfigValueImpl(key, null, null, null, 0));
+            return value;
         }
-        return findConfigValue("%" + configProfile + "." + key)
-                .or(() -> findConfigValue(key))
+
+        ConfigValue profileValue = findConfigValue("%" + configProfile + "." + key)
```

Comment:
here you could just use `orElse(value)`, and drop the `orElseGet()`

Reply:
Done